Add User Account dropin docs (#104)
commerce-docs/microsite-commerce-storefront: src/content/docs/dropins/user-account/styles.mdx

---

> title: User Account styles

Reviewer: ```suggestion
title: User account styles
```

> description: Learn how to override the styles used within the User Account drop-in.

Reviewer: ```suggestion
description: Learn how to override the styles used within the user account drop-in component.
```

---

import Aside from '@components/Aside.astro';
import Callouts from '@components/Callouts.astro';
import CodeImport from '@components/CodeImport.astro';

import accountLoaders from './files/AccountLoaders.css?raw';
import addressActions from  './files/AddressActions.css?raw';
import addressCard from './files/AddressCard.css?raw';
import addressFormWrapper from './files/AddressFormWrapper.css?raw';
import addressModal from './files/AddressModal.css?raw';
import changePassword from './files/ChangePassword.css?raw';
import customerInformationCard from './files/CustomerInformationCard.css?raw';
import editCustomerInformation from './files/EditCustomerInformation.css?raw';
import emptyList from './files/EmptyList.css?raw';
import form from './files/Form.css?raw';
import ordersListAction from './files/OrdersListAction.css?raw';
import ordersListCard from './files/OrdersListCard.css?raw';
import ordersListWrapper from './files/OrdersListWrapper.css?raw';
import Diagram from '@components/Diagram.astro';
import { Steps } from '@astrojs/starlight/components';

> This topic introduces you to the CSS classes for each UI component used in the User Account drop-in and shows you how to override these classes to customize the drop-in's CSS styling to match your brand.

Reviewer: ```suggestion
This topic introduces you to the CSS classes for each UI component used in the user account drop-in component and shows you how to override these classes to customize the user account component's CSS styling to match your brand.
```

:::tip[Boilerplate CSS Tip!]

> This drop-in contains multiple components that provide its UI. If you plan to make several CSS changes to the drop-in, we suggest creating CSS files for each drop-in component (in the `blocks/commerce-account/` directory) and importing those into the `commerce-account.css` file as shown below. This practice makes it easier to maintain and update your CSS in the future.

Reviewer: ```suggestion
User account contains multiple components that provide its UI. If you plan to make several CSS changes, Adobe suggests creating CSS files for each drop-in component (in the `blocks/commerce-account/` directory) and importing those into the `commerce-account.css` file as shown below. This practice makes it easier to maintain and update your CSS in the future.
```

```css title="commerce-account.css"
@import 'account-loaders.css';
@import 'address-actions.css';
@import 'address-card.css';
@import 'address-form-wrapper.css';
@import 'address-modal.css';
@import 'change-password.css';
@import 'customer-information-card.css';
@import 'edit-cuatomer-information.css';
@import 'empty-list.css';
@import 'form.css';
@import 'orders-list-action.css';
@import 'orders-list-card.css';
@import 'orders-list-wrapper.css';
```

:::

## Big Picture

> The quickest way to override User Account CSS is to inspect the User Account UI from your browser's developer tools to discover the BEM class names you want to add to or override. This process is numbered in the image below.

Reviewer: ```suggestion
The quickest way to override user account CSS is to inspect the user account UI from your browser's developer tools to discover the BEM class names you want to add to or override. This process is numbered in the image below.
```

<Diagram caption="Find CSS classes to override.">
  ![Find CSS classes to override](@images/dropins/user-account/find-styles-user-account.png)
</Diagram>

<Callouts columnCount="1">

1. **Inspect the element** in the UI that you want to customize (right-click on the element and select "Inspect" from the menu).
1. **Identify the CSS class(es)** for the element. We use [BEM naming](https://getbem.com/naming/), which makes it easy to know which component you're changing (and which CSS file to use).
1. **Copy the CSS class names to override** to your custom CSS file.

</Callouts>

## How to override the default styles

<Steps>

> 1. **Create a new CSS file** in your project for the User Account drop-in. Name the file `custom-account.css`.

Reviewer: ```suggestion
1. **Create a new CSS file** in your project for the user account component. Name the file `custom-account.css`.
```

1. **Copy the BEM class names** displayed in the Element tab of the developer panel into your new file.
1. **Add your custom CSS rules** to the `custom-account.css` file to add or override the default styles.
1. **Import the custom CSS file** into the `commerce-account.css` file.

   ```css
   @import 'custom-account.css';
   ```

1. **Save the file** and refresh your browser to see the changes.

</Steps>

## Example CSS overrides

> The following example shows how to override two of the many classes for the User Account drop-in component.

Reviewer: ```suggestion
The following example shows how to override two of the many classes for the user account component.
```

```css title="custom-account.css"
.checkout__heading {
  display: flex;
  justify-content: space-between;
}

.checkout__heading-title {
  line-height: 1.5;
  font-size: 1.5rem;
}
```

## Account CSS classes

> The CSS classes for each User Account drop-in component are provided here.

Reviewer: ```suggestion
The CSS classes for each user account component are provided here.
```
